src/php/datetime/strtotime.js

```diff
@@ -1,300 +1,1144 @@
+      this.h = relativeTo.getHours()
+    }
+
+    if (isNaN(this.i)) {
+      this.i = relativeTo.getMinutes()
+    }
+
+    if (isNaN(this.s)) {
+      this.s = relativeTo.getSeconds()
+    }
+
+    if (isNaN(this.f)) {
+      this.f = relativeTo.getMilliseconds()
+    }
+
+    // adjust special early
+    switch (this.firstOrLastDayOfMonth) {
+      case 1:
+        this.d = 1
+        break
+      case -1:
+        this.d = 0
+        this.m += 1
+        break
+    }
+
+    if (!isNaN(this.weekday)) {
+      var date = new Date(relativeTo.getTime())
+      date.setFullYear(this.y, this.m, this.d)
+      date.setHours(this.h, this.i, this.s, this.f)
+
+      var dow = date.getDay()
+
+      if (this.weekdayBehavior === 2) {
+        // To make "this week" work, where the current day of week is a "sunday"
+        if (dow === 0 && this.weekday !== 0) {
+          this.weekday = -6
+        }
+
+        // To make "sunday this week" work, where the current day of week is not a "sunday"
+        if (this.weekday === 0 && dow !== 0) {
+          this.weekday = 7
+        }
+
+        this.d -= dow
+        this.d += this.weekday
+      } else {
+        var diff = this.weekday - dow
+
+        // some PHP magic
+        if ((this.rd < 0 && diff < 0) || (this.rd >= 0 && diff <= -this.weekdayBehavior)) {
+          diff += 7
+        }
+
+        if (this.weekday >= 0) {
+          this.d += diff
+        } else {
+          this.d -= (7 - (Math.abs(this.weekday) - dow))
+        }
+
+        this.weekday = NaN
+      }
+    }
+
+    // adjust relative
+    this.y += this.ry
+    this.m += this.rm
+    this.d += this.rd
+
+    this.h += this.rh
+    this.i += this.ri
+    this.s += this.rs
+    this.f += this.rf
+
+    this.ry = this.rm = this.rd = 0
+    this.rh = this.ri = this.rs = this.rf = 0
+
+    let result = new Date(relativeTo.getTime())
+    // since Date constructor treats years <= 99 as 1900+
+    // it can't be used, thus this weird way
+    result.setFullYear(this.y, this.m, this.d)
+    result.setHours(this.h, this.i, this.s, this.f)
+
+    // note: this is done twice in PHP
+    // early when processing special relatives
+    // and late
+    // todo: check if the logic can be reduced
+    // to just one time action
+    switch (this.firstOrLastDayOfMonth) {
+      case 1:
+        result.setDate(1)
+        break
+      case -1:
+        result.setMonth(result.getMonth() + 1, 0)
+        break
+    }
+
+    // adjust timezone
+    if (!isNaN(this.z) && result.getTimezoneOffset() !== this.z) {
+      result.setUTCFullYear(
+        result.getFullYear(),
+        result.getMonth(),
+        result.getDate())
+
+      result.setUTCHours(
+        result.getHours(),
+        result.getMinutes() + this.z,
+        result.getSeconds(),
+        result.getMilliseconds())
+    }
+
+    return result
+  }
+}
+
+module.exports = function strtotime (str, now) {
+  //       discuss at: http://locutus.io/php/strtotime/
```

Comment:
I'd strongly suggest going with ES6 modules (using the easy-to-learn Rollup)... You can use Rollup to build CommonJS for Node, UMD for Node/Browser, IIFE for just browser, or even compile the source ES6 modules into ES6 modules which have references to external npm modules resolves into paths that the ES6-Module-supporting browsers understand (using plug-ins to convert from CommonJS, etc. if the external modules you are importing don't export ES6 modules themselves)--I don't know if php.js even has any external dependencies, in which case Rollup is even simpler...

Reply:
Oh, and you can add Babel to the Rollup compilation...